contrib/pinot-druid-migration/src/main/java/org/apache/pinot/druid/data/readers/DruidSegmentRecordReader.java

```diff
@@ -171,8 +171,8 @@ public GenericRow next(GenericRow reuse) {
         Object value = selector.getObject();
         if (value != null && !fieldSpec.isSingleValueField()) {
           // Multi-valued dimensions in Druid are stored as Arrays.ArrayList (this has been checked)
-          Preconditions.checkArgument(value.getClass() == java.util.Arrays.asList().getClass(), "The multi-valued dimension " + columnName +
-              " should be java.util.Arrays$ArrayList, but it is " + value.getClass());
+          Preconditions.checkArgument(value.getClass() == java.util.Arrays.asList().getClass(),
```

Comment:
As I mentioned in the previous comment, you need to use `equals()` for object comparison.
Moreover, you can try to use `instanceof` instead of checking class object by calling `getClass()`.
e.g. 
```
Preconditions.checkState(value instanceof List, errorMessage);
```